File: src/main/java/esport/sprint1/sprint1/models/Jeu.java
package esport.sprint1.sprint1.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonSetter;

import javax.persistence.*;
import java.io.Serializable;
import java.util.List;

@Entity
public class Jeu implements Serializable{

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ID")
    private Long id;

    private String nom;
    private String afficheUrl;
    private boolean type;

    @ManyToMany
    @JoinTable(
            name = "jeu_console",
            joinColumns = @JoinColumn(name = "JEU_ID", referencedColumnName = "ID"),
            inverseJoinColumns = @JoinColumn(name = "CONSOLE_ID", referencedColumnName = "ID")
    )
    private List<Console> consoles;

    @OneToMany(mappedBy = "jeu")
    private List<Tournois> tournois;

    public Jeu() {
    }

    public Jeu(String nom, String afficheUrl, boolean type) {
        this.nom = nom;
        this.afficheUrl = afficheUrl;
        this.type = type;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getAfficheUrl() {
        return afficheUrl;
    }

    public void setAfficheUrl(String afficheUrl) {
        this.afficheUrl = afficheUrl;
    }

    public boolean isType() {
        return type;
    }

    public void setType(boolean type) {
        this.type = type;
    }

    public List<Console> getConsoles() {
        return consoles;
    }

    public void setConsoles(List<Console> consoles) {
        this.consoles = consoles;
    }

    @JsonIgnore
    public List<Tournois> getTournois() {
        return tournois;
    }

    @JsonSetter
    public void setTournois(List<Tournois> tournois) {
        this.tournois = tournois;
    }
}

File: src/main/frontend/src/models/Jeu.ts
export class Jeu {
  private _id: number;
  private _Nom: string;
  private _AfficheUrl: string;
  private _Type: boolean;
  public constructor(
    _id: number,
    Nom: string,
    AfficheUrl: string,
    Type: boolean) {
    this._id = _id;
    this._Nom = Nom;
    this._AfficheUrl = AfficheUrl;
    this._Type = Type;
  }

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get Nom(): string {
    return this._Nom;
  }

  set Nom(value: string) {
    this._Nom = value;
  }

  get AfficheUrl(): string {
    return this._AfficheUrl;
  }

  set AfficheUrl(value: string) {
    this._AfficheUrl = value;
  }

  get Type(): boolean {
    return this._Type;
  }

  set Type(value: boolean) {
    this._Type = value;
  }
}
